modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqTranslate.scala

```diff
@@ -227,37 +227,40 @@ class SeqTranslate(site: Site, systems: Systems, settings: Settings) {
     else none
   }
 
-  def stopObserve(seqState: Sequence.State): Option[Process[Task, Event]] = deliverObserveCmd( seqState,
+  def stopObserve(seqId: Sequence.Id)(seqState: Sequence.State): Option[Process[Task, Event]] = deliverObserveCmd(
+    seqState,
     toInstrumentSys(seqState.toSequence.metadata.instrument).toOption.flatMap(_.observeControl match {
-      case Controllable(StopObserveCmd(stop), _, _, _) => Some(Process.eval(stop.run.map{
+      case Controllable(StopObserveCmd(stop), _, _, _, _, _) => Some(Process.eval(stop.run.map{
         case -\/(e) => Event.logMsg(SeqexecFailure.explain(e))
         case _      => Event.nullEvent
       }))
-      case _                                           => none
+      case _                                                 => none
     } )
-  )
+  ).orElse(stopPaused(seqId)(seqState))
```

Comment:
Basically, it tries both options. The state of the Observe Action determines which one will run.
This code should be improved.

Reply:
Improved how?